ports/stm32/pin_defs_stm32.c

```diff
@@ -8,7 +8,7 @@
 uint32_t pin_get_mode(const pin_obj_t *pin) {
     GPIO_TypeDef *gpio = pin->gpio;
     uint32_t mode = (gpio->MODER >> (pin->pin * 2)) & 3;
-    if (mode != GPIO_MODE_ANALOG) {
+    if (mode != GPIO_MODE_ANALOG && mode != GPIO_MODE_INPUT) {
```

Comment:
Would it make for clearer code if this were:
```c
if (mode == GPIO_MODE_OUTPUT_PP || mode == GPIO_MODE_AF_PP) {
...
```
?  Because they are the 2 modes where `OTYPER` modifies the output drive type.

Reply:
I was originally thinking that there was also GPIO_MODE_OUTPUT_OD and GPIO_MODE_OUTPUT_AD_OD which would make 4 things to compare against, but that's what we're constructing :)

So yeah - I agree it would be clearer to compare to the OUTPUT types. I'll update the PR